### PR TITLE
Implement https condition

### PR DESCRIPTION
**Pull request by carloosamarante**

### CODE REVIEW

File: n0s3p4ss/sniffer_switcher_http_status_based.py [hunk 1]
```diff
@@ -8,6 +8,14 @@
 from n0s3p4ss.sec_headers_obtainer import retrieve_x_frame_options, \
     retrieve_strict_transport_security, retrieve_content_security_policy, \
     retrieve_x_content_type_options
+from n0s3p4ss.ssl_analyser import SSLSocket, get_sslv23_method_context, \
```
Comment (cantoniazzi) [inline n0s3p4ss/sniffer_switcher_http_status_based.py]: the pep8 suggests that Imports should usually be on separate lines.
its like an `accent`, but I prefer 💃 
and the order should be
```
1 - Standard library imports.
2- Related third party imports.
3 - Local application/library specific imports.
```
ref. https://www.python.org/dev/peps/pep-0008/#imports <3

Comment (carloosamarante) [reply n0s3p4ss/sniffer_switcher_http_status_based.py]: Nice, i will change <3 !!

**Comment by carloosamarante** [comment]
I think it would be nice change some things, like tests and workflow... So if you guys have some tips, go ahed, i am not feeling so confortable with this current workflow :thinking:

File: n0s3p4ss/sniffer_switcher_http_status_based.py [hunk 2]
```diff
@@ -72,12 +80,50 @@ def apply(self, attack_surface):
...
+        if 'https' in attack_surface.response_url_location:
+            cert_info = self.certificate_info(attack_surface)
+            report['cert_info'] = cert_info
+        else:
+            alerts.append("HTTPS is not implemented")
 
         report['alerts'].extend(alerts)
         return report
 
+    def certificate_info(self, attack_surface):
```
Comment (julianoborba) [inline n0s3p4ss/sniffer_switcher_http_status_based.py]: This is a class method. Should be named as an action. Like get_certificate_info()

File: n0s3p4ss/sniffer_switcher_http_status_based.py [hunk 3]
```diff
@@ -72,12 +80,50 @@ def apply(self, attack_surface):
...
+        subdomain = attack_surface.domain
```
Comment (julianoborba) [inline n0s3p4ss/sniffer_switcher_http_status_based.py]: You can pass domain directly.

File: n0s3p4ss/sniffer_switcher_http_status_based.py (hunk 2 above)
Comment (rasecoiac03) [reply n0s3p4ss/sniffer_switcher_http_status_based.py]: I think the suggestion is good but `This is a class method.` is not a reason. :trollface:

Comment (julianoborba) [reply n0s3p4ss/sniffer_switcher_http_status_based.py]: Plase, help me with that. It's a function, so it had to be a behavior. Is this a good reason? Either way you are being a troll for not saying the right reason :trollface: .

**Pull request merged**